smart-contract-viem/README.md

```diff
@@ -0,0 +1,33 @@
+# Connect to Hedera Testnet via Viem
+
+This code snippets demonstrates how to connect to Hedera Testnet
+using Viem.
+
+## Code
+
+Copy the `.env.example` file to `.env`,
+then edit it to add the URL for a JSON-RPC endpoint for Hedera Testnet.
+
+To obtain this URL, see the [JSON-RPC connections documentation](https://docs.hedera.com/hedera/tutorials/more-tutorials/json-rpc-connections/).
+
+In the `hederaTestnet.js` file in this directory, a `hederaTestnetChain` is being created.
+This is essentially configuration for the RPC connection.
+
+The `HelloHedera.sol` file in this directory is the test solidity smart contract to be deployed/read from. `contract.js` is the related ABI.
+
+Subsequently a couple of RPCs can be invoked to deploy a contract (`deploy-contract.js`), read from a contract (`read-contract.js`) and subscribe to an event (`subscribe-to-event.js`).
+
+## Scripts
```

Comment:
also mention `contract.js` - which is `solc` compiler output. I assume that the reason that you have included this is to reduce the number of steps to reproduce - specifically the user does not need to install or runs `solc`, is this correct?